app/util/searchUtils.js

```diff
@@ -258,29 +265,48 @@ function getStops(input, origin) {
   )).then(suggestions => take(suggestions, 10));
 }
 
-export function executeSearchImmediate(getStore, { input, type }, callback) {
+export const getAllEndpointLayers = () => (
+  ['CurrentPosition', 'FavouritePlace', 'OldSearch', 'Geocoding']
+);
+
+
+export function executeSearchImmediate(getStore, { input, type, layers }, callback) {
   const position = getStore('PositionStore').getLocationState();
-  let endpoitSearches = [];
+  let endpointSearches = [];
   let searchSearches = [];
+  const endpointLayers = layers || getAllEndpointLayers();
 
   if (type === 'endpoint' || type === 'all') {
-    const origin = getStore('EndpointStore').getOrigin();
     const favouriteLocations = getStore('FavouriteLocationStore').getLocations();
     const oldSearches = getStore('OldSearchesStore').getOldSearches('endpoint');
     const language = getStore('PreferencesStore').getLanguage();
+    const searchComponents = [];
 
-    endpoitSearches = Promise.all([
-      getCurrentPositionIfEmpty(input, origin.useCurrentPosition),
-      getFavouriteLocations(favouriteLocations, input),
-      getOldSearches(oldSearches, input),
-      getGeocodingResult(input, position, language),
-    ])
+    if (endpointLayers.indexOf('CurrentPosition') !== -1 && position.hasLocation) {
```

Comment:
As we are using modern javascript, this could be done with Array.prototype.includes instead of Array.prototype.indexOf